kotlintest-core/src/main/kotlin/io/kotlintest/Spec.kt

```diff
@@ -110,4 +112,17 @@ fun Class<out Spec>.displayName(): String {
   }
 }
 
-fun Class<out Spec>.description() = Description.spec(this.displayName())
\ No newline at end of file
+fun Class<out Spec>.description() = Description.spec(this.displayName())
+
+val Spec.listenerInstances by LazyWithReceiver<Spec, List<TestListener>> { this.listeners() }
```

Comment:
Why use by LazyWithReceiver (what does it do)

Reply:
It's similar to lazy initialization, but I can call `this.*` inside it

It's an extension function for Spec, that guarantees that the listeners will be initialized only once. This was creating an issue when using listeners for the Environment/Properties, because the listener couldn't store state. It was initialized twice